source/base/bounding_box.cc

```diff
@@ -42,6 +42,86 @@ void BoundingBox<spacedim,Number>::merge_with(const BoundingBox<spacedim,Number>
 }
 
 template <int spacedim, typename Number>
+NeighborType BoundingBox < spacedim,Number >::is_neighbor(const BoundingBox<spacedim,Number> &other_bbox) const
+{
+  if (spacedim == 1)
+    {
+      // In dimension 1 if the two bounding box are neighbors
+      // we can merge them
+      if ( this->point_inside(other_bbox.boundary_points.first)
+           || this->point_inside(other_bbox.boundary_points.second) )
+        return mergeable_neighbors;
+      return not_neighbors;
+    }
+  else
+    {
+      std::vector< Point<spacedim,Number> > bbox1;
+      bbox1.push_back(this->get_boundary_points().first);
+      bbox1.push_back(this->get_boundary_points().second);
+      std::vector< Point<spacedim,Number> > bbox2;
+      bbox2.push_back(other_bbox.get_boundary_points().first);
+      bbox2.push_back(other_bbox.get_boundary_points().second);
+
+      // Step 1: testing if the boxes are close enough to intersect
+      for ( unsigned int d=0 ; d<spacedim; ++d )
+        if ( bbox1[0][d] - std::numeric_limits<Number>::epsilon() > bbox2[1][d]
```

Comment:
this epsilon check is a little weird, shouldn't it be relative? (if the value is larger than one, subtracting eps does nothing)

Reply:
Yes...this was completely wrong. Thank you for finding it!